lib/state/data/reducer.ts

```diff
@@ -553,3 +543,56 @@ export default combineReducers({
   preferences,
   tags,
 });
+
+const crossSliceReducer = (
+  state: DataState,
+  action: A.ActionType
+): DataState => {
+  switch (action.type) {
+    case 'RESTORE_NOTE_REVISION': {
```

Comment:
We could have this part in a separate function that only receives the required state slices (`notes` and `tags`).